client/state/reader/users/reducer.js

```diff
@@ -0,0 +1,48 @@
+import { combineReducers } from 'calypso/state/utils';
+import {
+	READER_USER_REQUEST,
+	READER_USER_REQUEST_SUCCESS,
+	READER_USER_REQUEST_FAILURE,
+} from '../action-types';
+
+// Stores the user data
+const items = ( state = {}, action ) => {
+	switch ( action.type ) {
+		case READER_USER_REQUEST_SUCCESS:
+			return {
+				...state,
+				[ action.userId ]: action.userData,
+			};
+		case READER_USER_REQUEST_FAILURE:
+			return {
+				...state,
+				[ action.userId ]: null,
+			};
+		default:
+			return state;
+	}
+};
+
+// Tracks loading states
+const requesting = ( state = {}, action ) => {
+	switch ( action.type ) {
+		case READER_USER_REQUEST:
+			return {
+				...state,
+				[ action.userId ]: true,
+			};
+		case READER_USER_REQUEST_SUCCESS:
+		case READER_USER_REQUEST_FAILURE:
+			return {
+				...state,
+				[ action.userId ]: false,
+			};
```

Comment:
Should we set this to `false` here, or clear it out once the request is done?

The alternative could look something like this:

```
case READER_USER_REQUEST_FAILURE:
			const { [action.userId]: _, ...newState } = state;
			return newState;
```

Reply:
I think setting it to false makes sense. It's consitent with other reducers and I think it makes it clear the request failed.